thavalon-webapp/src/App.tsx

```diff
@@ -13,21 +13,39 @@ import ReactModal from 'react-modal';
 ReactModal.setAppElement("#root");
 
 function App() {
+  // state for checking if logged in
   const [loggedIn, setLoggedIn] = useState(false);
+  // state for checking if logged in already checked via useEffect, to prevent it from briefly loading
+  // the register page before redirect to account page if user is logged in
+  const [checkedLoggedIn, setCheckedLoggedIn] = useState(false);
+  // state for checking if should display mobile navbar menu
   const [useMobileMenu, setUseMobileMenu] = useState(false);
+  // state for determining if login modal shown, controlled by navbar click
+  const [showLoginModal, setShowLoginModal] = useState(false);
+
   // check logged in status within useEffect to not enter render loop
   useEffect(() => {
     const accountManager = AccountManager.getInstance();
     accountManager.checkLoggedIn().then((httpResponse: HttpResponse) => {
       // calling set logged in will, on success, trigger a timer to regularly check refresh token
       setLoggedIn(httpResponse.result);
-    });  
+      setCheckedLoggedIn(true);
+    });
   })
 
+  // check if should redirect from register to account page
+  let registerPage: JSX.Element = <></>;
+  if (checkedLoggedIn) {
+    if (loggedIn) {
+      registerPage = <Redirect to="/account" />;
+    } else {
+      registerPage = <Register setLoggedIn={setLoggedIn} setShowLoginModal={setShowLoginModal} />;
+    }
+  }
```

Comment:
Up to you, but maybe we should make `registerPage` a function to pass directly to the `render` prop of `<Route>` below. That way, the components'll only get created if they're used